enterprise/tailnet/pgcoord.go

```diff
@@ -683,21 +715,136 @@ func (q *querier) newConn(c *connIO) {
 		return
 	}
 	cm.count++
-	q.conns[c] = struct{}{}
-	go q.cleanupConn(c)
+	q.conns[c.UniqueID()] = c
+	go q.waitCleanupConn(c)
 }
 
-func (q *querier) cleanupConn(c *connIO) {
-	<-c.ctx.Done()
+func (q *querier) newClientSubscription(c agpl.Queue, agentID uuid.UUID) error {
 	q.mu.Lock()
 	defer q.mu.Unlock()
-	delete(q.conns, c)
+
+	if _, ok := q.clientSubscriptions[c.UniqueID()]; !ok {
+		q.clientSubscriptions[c.UniqueID()] = map[uuid.UUID]struct{}{}
```

Comment:
this is a memory leak: you add to `q.clientSubscriptions` but never remove.

Reply:
You still never delete anything from `q.clientSubscriptions`

Reply:
Whoops, I think I got this now.